app/logic/Mail/ActiveSync/ActiveSyncFolder.ts

```diff
@@ -98,36 +102,46 @@ export class ActiveSyncFolder extends Folder implements ActiveSyncPingable {
       // If the function currently holding the sync key throws, we don't care.
     }
     try {
-      this.syncKeyBusy = this.makeSyncRequest(data);
+      this.syncKeyBusy = this.makeSyncRequest(data, responseFunc);
       return await this.syncKeyBusy;
     } finally {
       this.syncKeyBusy = null;
     }
   }
 
-  protected async makeSyncRequest(data?: any): Promise<any> {
-    let request = {
-      Collections: {
-        Collection: Object.assign({
-          SyncKey: this.syncState || "0",
-          CollectionId: this.id,
-        }, data),
-      },
-    };
-    let response = await this.account.callEAS("Sync", request);
-    if (!response) {
-      return null;
-    }
-    if (response.Collections.Collection.Status == "3") {
-      // Out of sync.
-      this.syncState = null;
+  /**
+   * Makes a `Sync` request to the server. It is called by
+   * `queuedSyncRequest()` and it may be called multiple times.
```

Comment:
Not `makeSyncRequest` may be called multiple times, but `responseFunc`

Reply:
same everywhere

Reply:
Sorry my bad, I was reading that wrong.